schutzbot/scheduled_cloud_cleaner.sh

```diff
@@ -84,3 +84,114 @@ for i in $(seq 0 $(("$STORAGE_ACCOUNT_COUNT"-1))); do
 done
 
 echo "Azure cleanup complete!"
+
+
+#---------------------------------------------------------------
+# 			AWS cleanup
+#---------------------------------------------------------------
+
+greenprint "Starting aws cleanup"
+
+TEMPDIR=$(mktemp -d)
+function cleanup() {
+    sudo rm -rf "$TEMPDIR"
+}
+trap cleanup EXIT
+
+# Check available container runtime
+if which podman 2>/dev/null >&2; then
+    CONTAINER_RUNTIME=podman
+elif which docker 2>/dev/null >&2; then
+    CONTAINER_RUNTIME=docker
+else
+    echo No container runtime found, install podman or docker.
+    exit 2
+fi
+
+CONTAINER_IMAGE_CLOUD_TOOLS="quay.io/osbuild/cloud-tools:latest"
+
+SSH_DATA_DIR=$(/usr/libexec/osbuild-composer-test/gen-ssh.sh)
+#SSH_KEY=${SSH_DATA_DIR}/id_rsa
+
+# We need awscli to talk to AWS.
+if ! hash aws; then
+    echo "Using 'awscli' from a container"
+    sudo ${CONTAINER_RUNTIME} pull ${CONTAINER_IMAGE_CLOUD_TOOLS}
+
+#----------------------------------
+#TODO: Do we need all this options?
+#----------------------------------
+
+    AWS_CMD="sudo ${CONTAINER_RUNTIME} run --rm \
+        -e AWS_ACCESS_KEY_ID=${V2_AWS_ACCESS_KEY_ID} \
+        -e AWS_SECRET_ACCESS_KEY=${V2_AWS_SECRET_ACCESS_KEY} \
+        -v ${TEMPDIR}:${TEMPDIR}:Z \
+        -v ${SSH_DATA_DIR}:${SSH_DATA_DIR}:Z \
+        ${CONTAINER_IMAGE_CLOUD_TOOLS} aws --region $AWS_REGION --output json --color on"
+else
+    echo "Using pre-installed 'aws' from the system"
+    AWS_CMD="aws --region $AWS_REGION --output json --color on"
+fi
+$AWS_CMD --version
+
+# Remove tagged and old enough instances
+INSTANCES=$(${AWS_CMD} ec2 describe-instances | jq -c '.Reservations[].Instances[]|try {"Tag": .Tags[],"InstanceId": .InstanceId,"LaunchTime": .LaunchTime}')
+
+for instance in ${INSTANCES}; do
+        TAG=$(echo "${instance}" | jq .Tag.Key)
+	TAG_VALUE=$(echo ${image} | jq .Tag.Value)
+        INSTANCE_ID=$(echo "${instance}" | jq .InstanceId)
+        LAUNCH_TIME=$(echo "${instance}" | jq .LaunchTime | tr -d '"')
+
+        if [[ ${TAG} =~ "gitlab-ci-test" && ${TAG_VALUE} =~ "true" ]]; then	
+		if [[ $(date -d "${LAUNCH_TIME}" +%s) -lt "${DELETE_TIME}" ]]; then
+        	        # Will be removed when reviewed
+			#$AWS_CMD ec2 terminate-instances --instance-id "${INSTANCE_ID}"
```

Comment:
Since the cloud-cleaner should only be ran in a scheduled pipeline which won't run this code until it gets merged it's fine to have it uncommeted.